### PR TITLE
Add more unit / integration tests for Swaps

### PR DESCRIPTION
**Pull request by dan437**
## Explanation
Add more unit / integration tests for Swaps

### CODE REVIEW

**Review by darkwing — DISMISSED** [review]

**Review by darkwing — APPROVED** [review]

File: jest.config.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ module.exports = {
   coverageThreshold: {
     global: {
       branches: 44,
-      functions: 42,
+      functions: 46.8,
```
Comment (adonesky1) [inline jest.config.js:14]: 🙌

File: ui/ducks/swaps/swaps.test.js [hunk 2]
```diff
@@ -288,6 +459,234 @@ describe('Ducks - Swaps', () => {
...
+        data: '0x095ea7b300000000000000000000000095e6f48254609a6ee006f7d493c8e5fb97094cef0000000000000000000000000000000000000000004a817c7ffffffdabf41c00',
+        to: '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2',
+        value: '0x0',
+        from: '0x2369267687A84ac7B494daE2f1542C40E37f4455',
+        gas: '0x12',
+        gasPrice: '0x34',
+      };
+      expect(swaps.getApproveTxParams(state)).toMatchObject({
+        ...state.metamask.swapsState.quotes.TEST_AGG_2.approvalNeeded,
+        gasPrice: 5,
```
Comment (adonesky1) [inline ui/ducks/swaps/swaps.test.js:685]: Why is the `gasPrice` different here?

File: ui/pages/swaps/awaiting-swap/awaiting-swap.js [hunk 3]
```diff
@@ -283,6 +283,7 @@ export default function AwaitingSwap({
       ) : null}
       <SwapsFooter
         onSubmit={async () => {
+          /* istanbul ignore next */
```
Comment (adonesky1) [inline ui/pages/swaps/awaiting-swap/awaiting-swap.js:286]: total nit: why not `/* istanbul ignore if */`

**Review by adonesky1 — APPROVED** [review]
LGTM!

File: ui/ducks/swaps/swaps.test.js (hunk 2 above)
Comment (dan437) [reply ui/ducks/swaps/swaps.test.js:685]: It's using the `fallBackPrice` prop in this case if an approval is needed: https://github.com/MetaMask/metamask-extension/blob/develop/test/jest/mock-store.js#L109

File: ui/pages/swaps/awaiting-swap/awaiting-swap.js (hunk 3 above)
Comment (dan437) [reply ui/pages/swaps/awaiting-swap/awaiting-swap.js:286]: That one would work as well.

**Pull request merged**